### DISCUSSION TITLE
ApplicationContext not working on win server 2016

### DISCUSSION THREAD

**Discussion by uwefms** [MarimerLLC/csla#1667] [Questions] [answered]
Hi everybody,

  I deployed the same server side Blazor app to two different webservers (2012 R2 and 2016). Both apps work fine except ApplicationContext.User, which has no data (.User.IsInRole()) in winserver 2016. Any ideas, what could be the reason ?

    regards

        Uwe

**Comment by rockfordlhotka** [comment, marked as answer]
Same version of .NET?

Are you using a custom principal/identity, or `ClaimsPrincipal` and `ClaimsIdentity`?

**Reply by uwefms** [reply]
Hi Rocky,

   sorry for taking so long to react. Finally found out the reason:
In Windows Features WebSocket Protokoll was not checked. Funny enough - everything was working except ApplicationContexr.User. I was using ClaimsPrincipal and ClaimsIdentity based on your Sample in ProjectTracker.

   Uwe
